### ISSUE TITLE
run kata-manager daily or on every PR/CI run?

### ISSUE THREAD

**Issue by jodh-intel** [kata-containers/kata-containers#6698]
The [`kata-manager.sh`](https://github.com/kata-containers/kata-containers/blob/main/utils/kata-manager.sh) install script is extremely useful for users, developers and testers who need a quick install to try out Kata, or just get a quick **static**  (not automatically updated!!) Kata installation.

However, it broke recently due to an upstream change to containerd. @amshinde has fixed the issue on #6695. However, I think we need to run the script either daily or on every PR/CI run to avoid it breaking silently again.

Thoughts folk?

**Comment by amshinde** [comment]
I agree @jodh-intel. I was going to suggest the same. Since we may point new users to the script, it needs to be tested. 
It could be implemented as github action. I believe you can schedule github actions to be run. 
It could also be triggered by any changes to the script itself.

**Comment by amshinde** [comment]
I forgot we would have to deal with nested virtualization as well.

**Comment by jodh-intel** [comment]
@amshinde - yep. Although GH's codespaces (and thus probably actions) environments _do_ appear to support nested virtualisation, it still won't work fwics. I tweaked the script to not require systemd, but it still fails since:

- containerd cannot mount overlay resources even though `overlay` appears in `/proc/modules`:
  ```bash
  $ grep overlay /proc/modules 
  overlay 114688 1 - Live 0x0000000000000000
  ```
  An `strace` of `ctr` shows:
  ```
  20526 mount("overlay", "/tmp/containerd-mount4185162833", "overlay", MS_RDONLY, "index=off,workdir=/var/lib/containerd/io.containerd.snapshotter.v1.overlayfs/snapshots/9/work,upperdir=/var/lib/containerd/io.containerd.snapshotter.v1.overlayfs/snapshots/9/fs,lowerdir=/var/lib/containerd/io.containerd.snapshotter.v1.overlayfs/snapshots/2/fs:/var/lib/containerd/io.containerd.snapshotter.v1.overlayfs/snapshots/1/fs" <unfinished ...>
  20526 <... mount resumed>)              = -1 EINVAL (Invalid argument)
  ```

- The `mod` tooling isn't available so we can't `modprobe vhost_vsock`. We could work around this issue by switching the config to use Cloud Hypervisor (hybrid VSOCK), but I don't think we can work around the containerd issue.

**Comment by jodh-intel** [comment]
I've also updated https://github.com/kata-containers/community/wiki/Outreach-preflight-checklist#system-specification with some of the details above.

**Comment by rootxrishabh** [comment]
Folks, Is this issue still available to work on? If yes, I would like to work on it. Is the nested virtualisation workaround available?
